### ISSUE TITLE
why getTransport use RLock to modify transports map instead of RWLock

### ISSUE THREAD

**Issue by zengxiaobai** [bfenetworks/bfe#553]
```
// getTransport return transport from map, if not exist, create a transport.
func (p *ReverseProxy) getTransport(cluster *bfe_cluster.BfeCluster) bfe_http.RoundTripper {
	p.tsMu.RLock()
	defer p.tsMu.RUnlock()

	transport, ok := p.transports[cluster.Name]
	if !ok {
		transport = createTransport(cluster)
		p.transports[cluster.Name] = transport
	}

	return transport
}
```

**Comment by iyangsj** [comment]
Thanks for your feedback!   Could you send a PR to fix this bug ?

```
	p.tsMu.RLock()
	transport, ok := p.transports[cluster.Name]
	p.tsMu.RUnlock()

	if !ok {
		transport = createTransport(cluster)
	        p.tsMu.Lock()
		p.transports[cluster.Name] = transport
                p.tsMu.Lock()
	}

	return transport
```

**Comment by iyangsj** [comment]
The issue is resolved by #563
